linux/src/file_util_linux.c

```diff
@@ -0,0 +1,135 @@
+                result_creation_disposition = O_CREAT;
+            }
+            else if (creation_disposition == CREATE_NEW)
+            {
+                /*Codes_SRS_FILE_UTIL_LINUX_09_016: [ If creation_disposition is CREATE_NEW and the file already exists, file_util_open_file shall fail and return INVALID_HANDLE_VALUE. ]*/
+                /*Codes_SRS_FILE_UTIL_LINUX_09_015: [ If creation_disposition is CREATE_NEW, file_util_open_file shall call open with O_CREAT|O_EXCL and shall return a new file handle if the file doesn't already exist. ]*/
+                result_creation_disposition = O_CREAT|O_EXCL;
+            }
+            else if (creation_disposition == TRUNCATE_EXISTING)
+            {
+                /*Codes_SRS_FILE_UTIL_LINUX_09_017: [ If creation_disposition is TRUNCATE_EXISTING, file_util_open_file shall call open with O_TRUNC and shall return a file handle who's size has been truncated to zero bytes. ]*/
+                result_creation_disposition = O_TRUNC;
+            }
+
+            int flags = user_access|result_creation_disposition;
+            result->h_file = open(full_file_name, flags);
+
+            if(result->h_file == -1)
+            {
+                /*Codes_SRS_FILE_UTIL_LINUX_09_008: [ If there are any failures, file_util_open_file shall fail and return INVALID_HANDLE_VALUE. ]*/
+                LogError("Failure in creating a file, full_file_name = %s, uint32_t access = %p, uint32_t share_mode = %p, LPSECURITY_ATTRIBUTES security_attributes = %p, uint32_t creation_disposition = %p, uint32_t flags_and_attributes = %p, HANDLE template_file = %p",
+                    full_file_name, access, share_mode, security_attributes, creation_disposition, flags_and_attributes, template_file);
+                free(result);
+                result = INVALID_HANDLE_VALUE;
+            }
+        }
+    }
+    /*Codes_SRS_FILE_UTIL_LINUX_09_020: [ file_util_open_file shall succeed and return a non-NULL value. ]*/
+    return result;
+}
+
+/*Codes_SRS_FILE_UTIL_LINUX_09_011: [ file_util_close_file shall close the given file handle. ]*/
+bool file_util_close_file(HANDLE handle_input)
+{
+    bool result;
+
+    if(handle_input == INVALID_HANDLE_VALUE)
+    {   
+        result = false;
+        LogError("Invalid argument to file_util_close_file. Handle input was an INVALID_HANDLE_VALUE");
+    }
+    else 
+    {
+        CREATE_FILE_LINUX* cfl = (CREATE_FILE_LINUX*)handle_input;
+
+        /*Codes_SRS_FILE_UTIL_LINUX_09_019: [ file_util_close_file shall call close on the given handle_input. ]*/
+        if(close(cfl->h_file) != 0)
+        {
+            result = false;
+            LogError("Failure in closing file");
+        }
+        else 
+        {
+            result = true;
+        }
+        free(cfl);
+    }
+
+    /*Codes_SRS_FILE_UTIL_LINUX_09_009: [ file_util_close_file shall fail and return false. ]*/
```

Comment:
>file_util_close_file shall fail and return false. [](http://example.com/codeflow?start=42&length=49)

If there are any failures, file_util_close_file shall fail and return false.

This should be placed in line 122